src/core/client_channel/client_channel_filter.cc

```diff
@@ -2213,12 +2213,15 @@ ClientChannelFilter::FilterBasedCallData::FilterBasedCallData(
       call_context_(args.context),
       call_start_time_(args.start_time),
       deadline_(args.deadline),
-      deadline_state_(
-          elem, args,
-          GPR_LIKELY(static_cast<ClientChannelFilter*>(elem->channel_data)
-                         ->deadline_checking_enabled_)
-              ? args.deadline
-              : Timestamp::InfFuture()) {
+      arena_(args.arena),
+      elem_(elem),
+      owning_call_(args.call_stack),
+      call_combiner_(args.call_combiner) {
+  if (GPR_LIKELY(static_cast<ClientChannelFilter*>(elem->channel_data)
+                     ->deadline_checking_enabled_)) {
+    static_cast<Call*>(call_context_[GRPC_CONTEXT_CALL].value)
+        ->UpdateDeadline(args.deadline);
```

Comment:
Why is this needed?  Won't the deadline timer have already been set in call.cc?

Reply:
Maybe... this is duplicating the logic that was in grpc_deadline_state.

Reply:
Looks like it's safe to remove so doing so.